destination/jetstream/writer.go

```diff
@@ -0,0 +1,183 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jetstream
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/nats-io/nats.go"
+)
+
+// message holds a message data and an ack function.
+type message struct {
+	data    []byte
+	ackFunc sdk.AckFunc
+}
+
+// Writer implements a JetStream writer.
+// It writes messages asynchronously.
+type Writer struct {
+	sync.Mutex
+
+	conn          *nats.Conn
+	subject       string
+	jetstream     nats.JetStreamContext
+	messages      []*message
+	batchSize     int
+	retryWait     time.Duration
+	retryAttempts int
+}
+
+// WriterParams is an incoming params for the NewWriter function.
+type WriterParams struct {
+	Conn          *nats.Conn
+	Subject       string
+	BatchSize     int
+	RetryWait     time.Duration
+	RetryAttempts int
+}
+
+// NewWriter creates new instance of the Writer.
+func NewWriter(ctx context.Context, params WriterParams) (*Writer, error) {
+	jetstream, err := params.Conn.JetStream()
```

Comment:
Did we do ping here?

Reply:
We are connecting to NATS [here](https://github.com/conduitio-labs/conduit-connector-nats/blob/destination-jetstream/destination/destination.go#L71), the `nats.Connect` will raise an error if a server is down or something wrong with the network.

But I can make a use of [Ping/Pong](https://docs.nats.io/using-nats/developer/connecting/pingpong). The Ping/Pong functionality implements periodic checks, to see if the client and server are still connected to each other. Do you think it would be good to check this periodically?

Reply:
Periodically ,  I don't think so, I see that's checking in open method it is enough
Thank you